### PR TITLE
Add a time sequence prediction example

### PR DESCRIPTION
**Pull request by maplewizard**
Hi,
I have made a time sequence prediction example, which I think, is capable to illustrate the usage of LSTMCell in pytorch. It's enough simple and fun and is a good small case for beginners who are interested in either pytorch or time sequence.

### CODE REVIEW

**Comment by maplewizard** [comment]
@soumith , I have removed the OpenNMT code, and this is a new PR.

File: time_sequence_prediction/generate_sine_wave.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import math
+import numpy as np
+import cPickle as pickle
```
Comment (soumith) [inline time_sequence_prediction/generate_sine_wave.py]: can you just use `torch.save` to dump these.

Also, import cPickle only works in python2 fyi

File: time_sequence_prediction/train.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
+from __future__ import print_function
+import torch
+import torch.nn as nn 
+from torch.autograd import Variable
+import cPickle as pickle
```
Comment (soumith) [inline time_sequence_prediction/train.py]: remove cPickle and just use `torch.load`

**Comment by maplewizard** [comment]
@soumith It has been replaced.

**Comment by soumith** [comment]
thank you @maplewizard for your contribution.

**Pull request merged**